contracts/managers/triggers/MovingAverageToAssetPriceCrossoverTrigger.sol

```diff
@@ -0,0 +1,239 @@
+     * @param  _signalConfirmationMaxTime           The amount of time, in seconds, until end of confirmation period
+     * @param  _initialState                        The trigger market state upond deployment
+     */
+    constructor(
+        IMetaOracleV2 _movingAveragePriceFeedInstance,
+        IOracle _assetPairOracleInstance,
+        uint256 _movingAverageDays,
+        uint256 _signalConfirmationMinTime,
+        uint256 _signalConfirmationMaxTime,
+        bool _initialState
+
+    )
+        public
+    {
+        // Set all state variables
+        movingAveragePriceFeedInstance = _movingAveragePriceFeedInstance;
+        assetPairOracleInstance = _assetPairOracleInstance;
+        movingAverageDays = _movingAverageDays;
+        signalConfirmationMinTime = _signalConfirmationMinTime;
+        signalConfirmationMaxTime = _signalConfirmationMaxTime;
+        lastConfirmedState = _initialState;
+        triggerFlippedIndex = 0;
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * If enough time has passed since last initial confirmation the current market state is
+     * calculated then compared to the last confirmed market state. If current state differs
+     * from last confirmed state then timestamp is logged to be used in calculating the start
+     * and end of the confirmation period.
+     */
+    function initialTrigger()
+        external
+    {
+        // Make sure enough time has elapsed since last initialTrigger
+        require(
+            hasConfirmationWindowElapsed(),
+            "MovingAverageToAssetPriceCrossoverTrigger.initialTrigger: Not enough time passed from last initial crossover."
+        );
+
+        // Get current market state and check that it's different from last confirmed state
+        bool currentMarketState = getCurrentMarketState();
+        require(
+            currentMarketState != lastConfirmedState,
+            "MovingAverageToAssetPriceCrossoverTrigger.initialTrigger: Market conditions have not changed since last confirmed state."
+        );
+
+        lastInitialTriggerTimestamp = block.timestamp;
+    }
+
+    /*
+     * If within the confirmation time period, the current market state is calculated then
+     * compared to the last confirmed market state. If current state differs from last confirmed
+     * state then the last confirmed state is updated to the current market state.
+     */
+    function confirmTrigger()
+        external
+    {
+        // Make sure currently in confirmation window
+        require(
+            inConfirmationWindow(),
+            "MovingAverageToAssetPriceCrossoverTrigger.confirmPropose: Confirming signal must be within bounds of the confirm propose."
+        );
+
+        // Get current market state and check that it's different from last confirmed state
+        bool currentMarketState = getCurrentMarketState();
```

Comment:
I'm cool having this be one line  with the require statement
```
getCurrentMarketState() != lastConfirmedState

```

Reply:
Yeah was doing that cause `currentMarketState` is used later but I'll just negate the `lastConfirmedState` when setting that variable